src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7040,7 +7044,9 @@ private BoundExpression BindArrayAccess(ExpressionSyntax node, BoundExpression e
                 ? arrayType
                 : arrayType.ElementType;
 
-            return new BoundArrayAccess(node, expr, convertedArguments.AsImmutableOrNull(), resultType, hasErrors);
+            return hasErrors
+                ? new BoundArrayAccess(node, BindToTypeForErrorRecovery(expr), convertedArguments.Select(e => BindToTypeForErrorRecovery(e)).AsImmutableOrNull(), resultType, hasErrors: true)
```

Comment:
>AsImmutableOrNull [](start = 141, length = 17)

```AsImmutable```? Otherwise if ```convertedArguments``` could be null, ```Select``` could crash.

Reply:
(this comment does not appear in CodeFlow)
That code is not changed from the version in the existing compiler.